### PR TITLE
Qual: Fix phan notifications for compta/facture/card

### PR DESCRIPTION
**Pull request by mdeweerd**
# Qual: Fix phan notifications for compta/facture/card

Disable ignores for compta/facture/card.php and fix notices.

### CODE REVIEW

**Comment by mdeweerd** [comment]
@eldy 
Only a few things left.

The main question about these lines where $remise_percent  and  $dateinvoice are undefined is: should the be fetched with GETPOST or not?

https://github.com/Dolibarr/dolibarr/blob/fcfd39b4ce9c36c65a3b3d214903442c8908bf41/htdocs/compta/facture/card.php#L3240-L3246

**Comment by mdeweerd** [comment]
@eldy I've ignored the PhanUndeclaredProperty notice again so that this PR can be integrated - I think that the remaining ones were introduced since I started this PR.

File: htdocs/compta/facture/card.php [hunk 1]
```diff
@@ -3297,7 +3302,7 @@
 				$classname = ucfirst($subelem);
 
 				$expesrc = new $classname($db);
-				'@phan-var-force CommonObject $expesrc';
+				'@phan-var-force Facture|Commande $expesrc';
```
Comment (eldy) [inline htdocs/compta/facture/card.php:3305]: I think the type of object in this case is Expedition

**Review by eldy — CHANGES_REQUESTED** [review]

**Comment by mdeweerd** [comment]
@eldy 
I added the dol_syslog statement to the code to log the class type so that you can check it yourself by visiting `compta/facture/card.php?action=create&origin=shipping&originid=2&socid=18` from the test database an check the log file contents.

**Pull request merged**